### ISSUE TITLE
Error when .value is used within computed property

### ISSUE THREAD

**Issue by gkaemmer** [danielearwicker/computed-async-mobx#6]
I discovered this nasty bug which I've reproduced here: https://jsfiddle.net/qc26pb4k/2/. With your console open, press "Show". This will trigger the computed being observed, and it will resolve. Then press "Hide". This unmounts a react component, which "unobserves" the computed async value.

It will throw a very cryptic mobx error `[mobx] Invariant failed: INTERNAL ERROR only onBecomeUnobserved shouldn't be called twice in a row`. It seems like it only happens when `.value` is used in a computed property. For some reason, unobserving the computed property triggers `onBecomeUnobserved` twice. Weird.

Also, I apologize for the copy/pasted computed-async-mobx code--it was the only way I could get it running in a fiddle.

(This might actually be a mobx bug, let me know what you think).

**Comment by gkaemmer** [comment]
This is definitely funky. Might be a subtle MobX bug. I wrote a test to make it easier to reproduce in development:
```
test("inComputed", (assert: test.Test) => {
    useStrict(strictness);

    const o = observable({ x: 0, y: 0 });
    const r = computedAsync<number>({
        init: 0,
        fetch: () => o.x + o.y
    });

    class Test {
        @computed get val() {
            return r.value;
        }
    }

    const t = new Test();

    // Observe the nested computed value
    const stop = autorun(() => t.val);

    assert.equal(t.rc, 0);

    // This throws an error
    stop();
});

**Comment by danielearwicker** [comment]
It seems to be trigged by my tearing down an internal `autorun` synchronously when my `atom` tells me to. If I delay this to a future event loop tick then it's okay:

```ts
    private sleep() {
        
        const monitor = this.monitor;
        this.monitor = undefined;
        
        if (monitor) {
            setTimeout(() => monitor, 1); // instead of just calling monitor();
        }
    }
```

However, this subtly changes the time at which various transitions occur, and my existing tests go haywire because they try to check every movement of the `busy` property. This is probably a lot fussier than they need to be.

So I need to rethink the tests entirely. I should be able to make them run a lot faster, for one thing, instead of using delay (which is bound to be indeterminate and flakey). It will probably take a few days for me to have much time to look at this though.

**Comment by gkaemmer** [comment]
Cool, that definitely fixes the problem for my use case. That solution seems a little hacky though--why can't mobx dispose the inner autorun synchronously?

**Comment by danielearwicker** [comment]
This is now fixed properly by the new implementation in 2.0.0.
